assignments/assignment7/faulty-oops.md

```diff
@@ -0,0 +1,84 @@
+
+`SET = 0, FnV = 0` indicates the syndrome exception type as synchronous exception. `FnV = 0` indicates the fault not valid bit as not set. `FnV` is a 1-bit field that indicates whether the exception is caused by a fault or not. `EA = 0, S1PTW = 0` indicates that the exception is not caused by an external abort. `S1PTW = 0` indicates stage 1 page table walk is not used.
+
+`FSC = 0x05: level 1 translation fault` indicates fault status code is of level 1 translation fault. `0x05` is the fault status code. `FSC` is a 6-bit field that indicates the type of fault that is raised. `level 1 translation fault` indicates fault of level 1 translation fault.
+
+`ISV = 0, ISS = 0x00000045` indicates invalid instruction specific syndrome. `0x00000045` is the instruction specific syndrome. `ISV` is a 25-bit field containining additional information about the exception that is raised.
+
+`CM = 0, WnR = 1` indicates invalid cache maintenance operation. `WnR = 1` indicates that the write not read bit is set. `WnR` is a 1-bit field indicating whether the exception was caused by a read or write operation.
+
+`user pgtable: 4k pages, 39-bit VAs, pgdp=0000000042079000` indicates user page table is using 4k pages and 39-bit virtual addresses. `pgdp=0000000042079000` indicates page global directory pointer is `0000000042079000`.
+
+`[0000000000000000] pgd=0000000000000000, p4d=0000000000000000, pud=0000000000000000` indicates invalid page global directory (pgd), invalid page 4th level directory (p4d), and invalid page upper directory (pud).
+
+`Internal error: Oops: 96000045 [#1] SMP` indicates kernel has raised an internal error. `96000045` is the ESR value. `#1` is the number of times the kernel has raised the internal error. `SMP` indicates kernel is running in a symmetric multiprocessing (SMP) configuration.
+
+`Modules linked in: hello(O) scull(O) faulty(O) [last unloaded: hello]` indicates `hello`, `scull`, and `faulty` modules are loaded. `hello` module is the last module to be unloaded.
+
+`CPU: 0 PID: 158 Comm: sh Tainted: G           O      5.15.18 #1` indicates CPU type is CPU 0, the process ID is 158, command is `sh`, and the kernel is tainted. `G` indicates kernel is running in a guest virtual machine. `O` indicates kernel is running in an out-of-tree configuration. `5.15.18` is the kernel version. `#1` is the kernel build number.
+
+`Hardware name: linux,dummy-virt (DT)` indicates the hardware name is `linux,dummy-virt` and the device tree is used.
+
+`pstate: 80000005 (Nzcv daif -PAN -UAO -TCO -DIT -SSBS BTYPE=--)` indicates that the processor state is `80000005`. `Nzcv daif -PAN -UAO -TCO -DIT -SSBS BTYPE=--` is the processor state flags. 
+
+`pc : faulty_write+0x14/0x20 [faulty]` indicates the program counter is `faulty_write+0x14/0x20 [faulty]`. `faulty_write` is the function that raised the exception. `0x14` is the offset from the start of the function. `0x20` is the size of the function. `faulty` is the module that contains the function.
```

Comment:
@TejaswiniLakshminarayan nice! the only detail I was looking for was this last line, but this analysis is very thorough!